File: ts/src/router/sync_route_handler.ts
import {RouteHandler} from './route_handler';
import {Promise, PromiseWrapper} from 'angular2/src/facade/async';
import {Type} from 'angular2/src/facade/lang';

export class SyncRouteHandler implements RouteHandler {
  /** @internal */
  _resolvedComponent: Promise<any> = null;

  constructor(public componentType: Type, public data?: {[key: string]: any}) {
    this._resolvedComponent = PromiseWrapper.resolve(componentType);
  }

  resolveComponentType(): Promise<any> { return this._resolvedComponent; }
}

File: lib/src/router/sync_route_handler.dart
library angular2.src.router.sync_route_handler;

import "route_handler.dart" show RouteHandler;
import "package:angular2/src/facade/async.dart" show Future, PromiseWrapper;
import "package:angular2/src/facade/lang.dart" show Type;

class SyncRouteHandler implements RouteHandler {
  Type componentType;
  Map<String, dynamic> data;
  /** @internal */
  Future<dynamic> _resolvedComponent = null;
  SyncRouteHandler(this.componentType, [this.data]) {
    this._resolvedComponent = PromiseWrapper.resolve(componentType);
  }
  Future<dynamic> resolveComponentType() {
    return this._resolvedComponent;
  }
}
